### DISCUSSION TITLE
[Idea] The new checkout uses a wrong strategy, penalizes sales by adding uncertainty to the purchase phase

### DISCUSSION THREAD

**Discussion by Maofree** [PrestaShop/PrestaShop#32250] [Ideas]
Hi
with the new checkout there is a big problem, very often people stop at the first step, registering only username and password without continuing in the next steps. The main problem is that if the steps of the various purchase phases are hidden, it certainly does not encourage people to complete all the information, they often get stuck without continuing. Why not use a checkout with open steps as they are present in the modules you addons.
In the end I don't understand these new solutions, I don't think it's so difficult to understand things customers want.
Using these additional modules adds compatibility problems with payment modules, courier modules, e-invoicing modules and management costs, as well as new errors.
I hope that in the next 1.8 certain solutions will follow a more standard and sensible logic.
I already know that in 1.7 you will not make changes.

this should be a good solution
https://addons.prestashop.com/it/express-checkout/42005-the-checkout.html
I do not understand why going to use a solution where the only strategy is to channel the customer to the purchase, neglecting the fears of customers, especially those who make the first purchase. The best solution is to make the whole process transparent and visible just to instill confidence and encourage them to purchase, without having to hide anything

so I need to pay 200€ for a single module and if I support 20 companies I should spend 4000€ for what? for a bad strategory made from Prestashop?
you should do regular solutions, and only if I want to try different solution I'll buy it from Addons
you should meet the demands of shopkeepers who depend on computer systems, and you should be more responsible for making choices, some including this, because in some cases they may be okay

on 1.7 there is another bad solution:
- each combinations have an url (too many errors there are with this solution)

all the rest is fine, but these 2 things are absurd, they are pure experiments that can be good only in some cases, instead you have made them by default creating problems and going against the current. Do not respond with the usual banal phrases caring about the consequences of certain decisions. This checkout is really a stupid solution.
I'm really angry, because I saw that it decreases the number of orders (don't come and tell me that it can depend on many factors, because I know the data of the companies I follow and I see a large number of people who register without adding addresses and now they are forced to waste time buying checkout modules on addons)

bye

**Screenshots**
![11](https://user-images.githubusercontent.com/653230/64273325-578a0e00-cf41-11e9-85a8-92855f5ce4d9.png)

**Additional information**
PrestaShop version: 1.7.6.1
PHP version: 7.0

**Comment by khouloudbelguith** [comment]
Hi @maofree,

This behavior is fixing some issues:
https://github.com/PrestaShop/PrestaShop/issues/12456
https://github.com/PrestaShop/PrestaShop/issues/13126
https://github.com/PrestaShop/PrestaShop/issues/14846
https://github.com/PrestaShop/PrestaShop/issues/11750

Thanks to check & feedback

**Comment by Maofree** [comment]
Hi
sorry but I speak about the strategy used in the new checkout not about a bug
I want all steps opened without the continue button. All blocks should be visible not hide like now.
The current solution does not inspire confidence in the purchase at all and many companies tell me that there are many more abandoned carts and those that register do not enter address data. In my opinion Prestashop has reflected little on the consequences of certain solutions. The problem is that then Prestashop doesn't change strategy quickly

**Comment by khouloudbelguith** [comment]
@maofree, thanks for your feedback.
Yes, this strategy is used to fix some issues on the check page as I said before.
But, I will ping our Product team to discuss about this new behaviour
Ping @PrestaShop/prestashop-product-team what do you think? 

Thanks!

**Comment by Maofree** [comment]
thanks for listening to me

**Comment by rdy4ever** [comment]
If you ask me, "Order with an obligation to pay" button scares more customers than this kind of checkout process. The text on the button should be just "Complete order" or "Finish order". The term "obligation" has no place here.
As for this kind of checkout process, it's just a step-by-step checkout that is also used by other e-commerce solutions for a while now.

**Comment by LouiseBonnard** [comment]
@rdy4ever, I admit this wording sounds a bit frightening but it might be there for legal purposes... I'll check with the legal team before and let you know! From my point of view, _Place order_ can be a nice alternative. :-)

**Comment by rdy4ever** [comment]
@LouiseBonnard I’m pretty sure you cannot force anyone to pay the order just because they use your check-out - legally speaking. Untill the very last moment when they actually pay for the merchandise (cash, card , bank wire, etc) they are at the liberty to default on the payment and/or refuse to receive the order. And if they have any doubt about their rights, that text on the button will only make them go away.

**Comment by LouiseBonnard** [comment]
@rdy4ever, @marionf, it turns out we can modify this wording, I've checked the legal team and it is not mandatory; the only requirement is that the client must be aware of his/her obligation to pay but it doesn't have to be written inside the button, it can be displayed in the terms & conditions pages. But that's to be precised on a merchant's side, not ours.

I suggest _Place order_ (localized in both _Shop.Theme.Actions_ and _Shop.Theme.Checkout_ domains) then! If it suits you as well, I'll create an issue so that it can be taken into account for the next version.

cc @colinegin

**Comment by Maofree** [comment]
Hi
however this problem on that button can be real but it appears at the end of the steps, the problem for me is at the begin, because now this checkout has increased the number of customers that finish their registration at the first step adding an account with only name, surname, email and password, without addresses. These steps should be visible to avoid doubts and fears

bye

**Comment by LouiseBonnard** [comment]
Dear @maofree, thanks for the issue, we did take note of your remark, be sure we will study and discuss it in the next few days, and keep you posted!

**Comment by LouiseBonnard** [comment]
Hello @maofree, to keep you posted, we've just talked about it. We do take note of this feedback, but the checkout is a big topic and it makes no sense to change one part without reworking the whole process. In other words, it won't be studied in the next few weeks but when we will decide to rework the checkout.

Meanwhile, I invite you to turn this into a suggestion in the [User Voice](http://feedback.prestashop.com/forums/387864-prestashop-1-7-x) so that other users can vote for your idea if they consider it important for their business as well, it will help our team estimate how demanded by the community is this feature.

**Comment by Maofree** [comment]
Hi @LouiseBonnard 
I know that it is big change, but it is not necessary that I ask for some votes, I've no time to do that. It is not so difficult to understand the motive to change the strategy about the checkout, on addons all checkout are with visible blocks, why do they do that? for the same motive, it is necessary to instilling confidence in customers without hiding anything from them because they are all fearful of taking scams.
I use and develop modules for prestashop by the version 1.2.5 and I help 20 companies so all of them don't like this new solution.
if you are a customer which solution do you prefer? with hidden steps or with visible blocks?
normally the second one, so by default you should offer the best solutions for each type of shop, and if somebody wants strange solution he can buy it from addons but you can't by default set an extreme solution right on the most important page of the sales site

bye

**Comment by Maofree** [comment]
create a new idea
http://feedback.prestashop.com/forums/387864-prestashop-1-7-x/suggestions/38736217-make-the-steps-on-the-checkout-page-visible

**Comment by matks** [comment]
@maofree @rdy4ever I would like to fully understand this issue.

If I am not wrong, you can build 1-step-checkout today with PS 1.7 ? PrestaShop does not prevent you from doing it.

The Classic Theme does not provide it but I think https://github.com/PrestaShopCorp/ps_checkout even provides 1-clic checkout 🤔  .

So if I get it correctly, it's only a limitation of Classic Theme?

**Comment by grzegorzgit** [comment]
> 
> 
> @rdy4ever, @marionf, it turns out we can modify this wording, I've checked the legal team and it is not mandatory; the only requirement is that the client must be aware of his/her obligation to pay but it doesn't have to be written inside the button, it can be displayed in the terms & conditions pages. But that's to be precised on a merchant's side, not ours.
> 
> I suggest _Place order_ (localized in both _Shop.Theme.Actions_ and _Shop.Theme.Checkout_ domains) then! If it suits you as well, I'll create an issue so that it can be taken into account for the next version.
> 
> cc @colinegin

@LouiseBonnard it is actually mandatory for the button to say something that will clearly tell the client that right now he is making an order that obligates him to pay. Despite of your return policy, there has to be a moment when the order is done - and by this the customer obligates himself to pay for it.  And because in many stores there are plenty buttons during the order - like in prestashop, there are 4 steps - there has to be an clear information to the user "by clicking this last button you are legally placing the order". But this applies only in EU, probably not elsewhere.

**Comment by matks** [comment]
Ping @eshraw and @MatShir I just remembered this issue and I think in 2022 it rings a new bell

**Comment by Maofree** [comment]
HI @matks 
in these months I can tell you what's wrong with this checkout.
The biggest problem with multiple customers is the shipping step, when multiple addresses are added. In practice, what creates problems for customers is understanding that they have to choose the shipping and billing address. If there are 2 addresses they see 4 addresses, 2 to choose for shipping and 2 to choose for billing.
A simpler or more intuitive selection would be needed, unfortunately there are still many people who are impractical with PCs and online shopping.
Unfortunately, when they find these difficulties, many abandon their purchases, only a few notify the company for help

bye

**Comment by korberg** [comment]
I think this topic needs attention, just migrated a shop to PrestaShop and have been looking ways to remove abundant fields.
The arguments for checkout rehaul are:
1) Discouragement of checkout with closed steps, so single-step checkout would benefit all users.
2) At the same time no simple way to custom add and require fields that YOUR shop's shipping/payment needs (this form builder in Admin is great, but I could not add phone as required field from there, I had to go to php for that). Eg. In our country, most of the sales are made by direct banklink payment (client will be redirected to bank and back to shop after payment is done) - thus there is no need for zip, address, city to ensure the authenticity of the client. If there are payment providers that NEED that info, it can then be included, otherwise easy way to opt out of those fields and personal data.
3) Almost ALL of shipping is done here from parcel-terminal to parcel-terminal. There is absolutely NO NEED for personal address, zip, even city. Let client fill out the address WHEN shipping/carrier needs it (as with payment), but if client will select from carrier page parcel terminal in desired city.... Why do one needs to give the shop personal address? If there is a personal courier option in some addon/module, let that address be included right there - if that selection is made. 

With the newer regulations in personal data, I think that rather than add reassurance module that says something like "Your information is safe with us" DO NOT REQUIRE more information than is needed.

I am working on workaround about that address issue currently as I know that clients are not willing to fill out form with unnecessary data (and also - I do not want to have that data!). As much as I have discovered, I cannot remove requirement for address1 field, but I hope I can hide it and prefill it with blank. 

And when all those unnecessary input fields are removed, it is even sillier to have multiple steps just to ask name in one, phone and country in the second and let select destination parcel terminal in the third/fourth. 

I will check out this paypal+presta checkout module, but as far as I know, Paypal wants more information and has been unwilling to proceed without zip/city in some instances, so not really counting on it :)

If there are any other currently-in-dev modules that I should be checking out, I would be happy to do so.
